packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -194,21 +193,11 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
 
   async deployConfig({
     apiKey,
-    developerPlatformClient,
     appConfiguration,
   }: ExtensionDeployConfigOptions): Promise<{[key: string]: unknown} | undefined> {
-    if (this.isFunctionExtension) return this.functionDeployConfig({apiKey, developerPlatformClient, appConfiguration})
     return this.commonDeployConfig(apiKey, appConfiguration)
```

Comment:
This is looking good!!!

I'll ask for one more change, since now `deployConfig` is basically doing nothing and just calling `commonDeployConfig` can we unify both?

Have a single `deployConfig` that does whatever `commonDeployConfig` does.
And update all calls to `commonDeployConfig` to call `deployConfig` instead. Should be straightforward :)